### PR TITLE
allow complex build for multideterminants with xml input

### PR DESCRIPTION
**Pull request by camelto2**
Please review the [developer documentation](https://github.com/QMCPACK/qmcpack/wiki/Development-workflow)
on the wiki of this project that contains help and requirements.

## Proposed changes
This is a simple fix to allow the multideterminant code to actually detect the CI coefficients from xlm input. Currently, all CI coefficients would be found as zero, because the ci coefficient was set before a put for the real and imaginary parts, so it was never correctly set. 

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- Bugfix

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
my workstation

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted

### CODE REVIEW

File: src/QMCWaveFunctions/Fermion/SlaterDetBuilder.cpp [hunk 1]
```diff
@@ -1197,6 +1198,9 @@ bool SlaterDetBuilder::readDetList(xmlNodePtr cur,
         confAttrib.add(tag, "id");
         confAttrib.put(cur);
 
+#ifdef QMC_COMPLEX
+        ValueType ci(ci_real, ci_imag);
```
Comment (ye-luo) [inline src/QMCWaveFunctions/Fermion/SlaterDetBuilder.cpp:1202]: I could not understand why this change makes a difference. Could you explain a bit more?

Comment (camelto2) [reply src/QMCWaveFunctions/Fermion/SlaterDetBuilder.cpp:1202]: inside the ifdef QMC_COMPLEX, ci_real and ci_imag aren't actually being updated until the confAttrib.put(cur) is called. So where the call was previously, the ci is being initialized as 0+i0 instead of the desired ci_real + i ci_imag. The ValueType ci(ci_real,ci_imag) has to happen after ci_real and ci_imag are actually read in

Comment (camelto2) [reply src/QMCWaveFunctions/Fermion/SlaterDetBuilder.cpp:1202]: Notice in the code above (around line 1000) when usingCSF is true, the correct behavior is obtained because ValueType ci(ci_real, ci_imag) is called after the put. This only effects the case when usingCSFs is false

Comment (ye-luo) [reply src/QMCWaveFunctions/Fermion/SlaterDetBuilder.cpp:1202]: I see it now. Thanks.

**Review by ye-luo — APPROVED** [review]

**Comment by ye-luo** [comment]
Okay to test
